src/PowerShellEditorServices.Protocol/LanguageServer/GetCommandsRequest.cs

```diff
@@ -0,0 +1,54 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using Microsoft.PowerShell.EditorServices.Protocol.MessageProtocol;
+using System.Collections.Generic;
+using System.Management.Automation;
+
+namespace Microsoft.PowerShell.EditorServices.Protocol.LanguageServer
+{
+    /// <summary>
+    /// Defines a class that describes the request to get all PowerShell Commands from the current session
+    /// </summary>
+    public class GetAllCommandsRequest
+    {
+        public static readonly
+            RequestType<List<PSAllCommandsMessage>, object, object, object> Type =
+            RequestType<List<PSAllCommandsMessage>, object, object, object>.Create("powerShell/getAllCommands");
+    }
+
+    /// <summary>
+    /// Defines a class that describes the request to get the details for a single PowerShell Command
+    /// from the current session.
+    /// </summary>
+        public class GetCommandRequest
+    {
+        public static readonly
+            RequestType<string, object, object, object> Type =
+            RequestType<string, object, object, object>.Create("powerShell/getCommand");
```

Comment:
So I just thought of something and I REALLY HOPE @corbob doesn't hate me after this one...

if the difference between PSCommandMessage and PSAllCommandsMessage is just a `DefaultParameterSet` and `ParameterSets` property

and the difference between 
"powerShell/getAllCommands"
and 
"powerShell/getCommand"

is just using the `param` object... could we combine these into 1 message?

`"powerShell/getCommand"` that... if param is empty, will do `Get-Command` and if it's not empty will do `Get-Command foo`?

I know this is a refactor... but I hope that my comment makes sense that it would be nice to have 1 message and one data type rather than 2 that do similar things.

Reply:
pls don't hate me

Reply:
Pretty sure this is taken care of by removing GetAllCommands. Going to mark it as resolved and hope for the best 🤣